Calendar view

joenas opened the pull request:
TODO:

- [x] Translate `CALENDAR_NO_CONTENT` and `CALENDAR_FULL_DAY`
- [x] Should deps be in `guide-app/package.json` or not?
- [x] Add new strings to Lund
- [x] Analytics


Regarding dependencies we had a discussion and agreed on **not** having them in `guide-app/package.json`, mostly because it's confusing. 
An argument would be to only have "linkable" packages in the actual cities and the "pure JS" ones in `guide-app` but that could easily be more confusing and involves just as much work, if not more, so skipping that.

Commit:
Get json-schemas from guide-app
Removed them from cities and template

Commit:
First stab at CalendarView with events from API

Commit:
Added date-fns and DateUtils

Commit:
Added DatePicker to CalendarView

Commit:
Extracted components from CalendarScreen

Commit:
Added no content info to CalendarScreen

Commit:
Correct date addition/subtraction in CalendarScreen

Commit:
Disabled previous date btn when today.
Minor design tweaks

Commit:
More date-fns

worldeggplant commented on guide-app/src/components/shared/CalendarDatePicker.js:
```diff
@@ -0,0 +1,66 @@
+type Props = {
+  chosenDate: Date,
+  currentLanguage: string,
+  // TODO: :)
```
TODO
Please document what to do if leaving it in.

joenas replied:
Fixeed

worldeggplant commented on guide-app/src/components/shared/CalendarEvent.js:
```diff
@@ -0,0 +1,156 @@
+    paddingHorizontal: 15,
+    shadowRadius: 12,
+    shadowColor: "black",
+    shadowOpacity: 0.16
```
You're not including Android's `elevation` prop here.

worldeggplant commented on guide-app/src/components/shared/CalendarEvent.js:
```diff
@@ -0,0 +1,156 @@
+  const image = imageUrl ? { uri: imageUrl } : defaultImage;
+
+  const entities = new XmlEntities();
+  const decodedLocationTitle = entities.decode(location.title);
```
Do we want to do the decoding in this component, or move this out & elsewhere (like in a reducer?)? Seems like something more tied to the API than this particular component.

joenas replied:
Hm yeah. Is the reducer the best place you think?

worldeggplant commented on guide-app/src/reducers/eventReducer.js:
```diff
@@ -0,0 +1,36 @@
+      //     // push
+      //     items.push(g);
+      //   }
+      // });
```
Comment

joenas replied:
Yeah this was one of the things I wanted to have a discussion about, but for now I think I'll just remove it and we can revisit

joenas replied:
Added some comments :)

Commit:
Merge branch 'develop' into topic/calendar-view

Commit:
Fixed typings and margins in CalendarDatePicker

Commit:
Fixed formatting of date in DatePicker, display entire month

Commit:
PR fixes: added elevation to style

Commit:
Added a bit of context to TODO regarding grouping by dates

Commit:
Moved dependencies from guide-app to cities/template

Commit:
Added CALENDAR_NO_CONTENT string to Lund and template

Commit:
Display text when an Event is full day

Commit:
Android fixes for Calendar:
Use white status bar
Less letterSpacing because Android support is a bit weird and affects layout. See RN docs

Commit:
Timestamp needs to be sent as `date` param to calendar webpage to display correct occasion

Commit:
Added missing calendar translations

Commit:
Added analytics to calendar view. Also fixed bug in AnalyticsUtils where parameters to firebase had changed

Commit:
Added paddingTop on CalendarPicker for Android, to align with new home screen
